src/jdk.compiler/share/classes/com/sun/tools/javac/jvm/Gen.java

```diff
@@ -230,8 +240,14 @@ Symbol binaryQualifier(Symbol sym, Type site) {
                 sym.owner != syms.arrayClass)
                 return sym;
             // array clone can be qualified by the array type in later targets
-            Symbol qualifier = new ClassSymbol(Flags.PUBLIC, site.tsym.name,
-                                               site, syms.noSymbol);
+            if (qualifiedSymbolCache == null) {
+                qualifiedSymbolCache = new HashMap<>();
```

Comment:
given that the map is initialized in the constructor I assume that this line should never be executed right? also I think that  you can clear the map in the `finally` block of method `Gen::genClass`

Reply:
I assumed that other developer will revise `qualifiedSymbolCache` elsewhere in the future. So I judge if it is null to avoid NPE.  The newly added code in the finally block of `Gen::genClass`  states that this judgement is necessary.